mobile/examples/object_detection/ios/ORTObjectDetection/ModelHandler.swift

```diff
@@ -0,0 +1,442 @@
+            first.score > second.score
+        }
+        
+        return resultsArray
+    }
+    
+    // This method preprocesses the image by cropping pixel buffer to biggest square and scaling the cropped image to model dimensions.
+    private func preprocess(
+        ofSize size: CGSize,
+        _ buffer: CVPixelBuffer
+    ) -> CVPixelBuffer? {
+        let imageWidth = CVPixelBufferGetWidth(buffer)
+        let imageHeight = CVPixelBufferGetHeight(buffer)
+        let pixelBufferType = CVPixelBufferGetPixelFormatType(buffer)
+        
+        assert(pixelBufferType == kCVPixelFormatType_32BGRA ||
+            pixelBufferType == kCVPixelFormatType_32ARGB)
+        
+        let inputImageRowBytes = CVPixelBufferGetBytesPerRow(buffer)
+        let imageChannels = 4
+        
+        CVPixelBufferLockBaseAddress(buffer, CVPixelBufferLockFlags(rawValue: 0))
+        
+        defer { CVPixelBufferUnlockBaseAddress(buffer, CVPixelBufferLockFlags(rawValue: 0)) }
+        
+        // Find the biggest square in the pixel buffer and advance rows based on it.
+        guard let inputBaseAddress = CVPixelBufferGetBaseAddress(buffer) else {
+            return nil
+        }
+        
+        // Get vImage_buffer
+        var inputVImageBuffer = vImage_Buffer(data: inputBaseAddress,
+                                              height: UInt(imageHeight),
+                                              width: UInt(imageWidth),
+                                              rowBytes: inputImageRowBytes)
+        
+        let scaledRowBytes = Int(size.width) * imageChannels
+        guard let scaledImageBytes = malloc(Int(size.height) * scaledRowBytes) else {
+            return nil
+        }
+                
+        var scaledVImageBuffer = vImage_Buffer(data: scaledImageBytes, height: UInt(size.height), width: UInt(size.width), rowBytes: scaledRowBytes)
+        
+        // Perform the scale operation on input image buffer and store it in scaled vImage buffer.
+        let scaleError = vImageScale_ARGB8888(&inputVImageBuffer, &scaledVImageBuffer, nil, vImage_Flags(0))
+        
+        guard scaleError == kvImageNoError else {
+            free(scaledImageBytes)
+            return nil
+        }
+        
+        let releaseCallBack: CVPixelBufferReleaseBytesCallback = { _, pointer in
+            
+            if let pointer = pointer {
+                free(UnsafeMutableRawPointer(mutating: pointer))
+            }
+        }
+        
+        var scaledPixelBuffer: CVPixelBuffer?
+        
+        // Convert the scaled vImage buffer to CVPixelBuffer
+        let conversionStatus = CVPixelBufferCreateWithBytes(
+            nil, Int(size.width), Int(size.height), pixelBufferType, scaledImageBytes,
+            scaledRowBytes, releaseCallBack, nil, nil, &scaledPixelBuffer
+        )
+        
+        guard conversionStatus == kCVReturnSuccess else {
+            free(scaledImageBytes)
+            return nil
```

Comment:
nit: is there a way to avoid the multiple calls to `free(scaledImageBytes)`?

Reply:
unfortunately, it seems like we have to manually call `free(scaledImageBytes)` at every exits..